### PR TITLE
DCOS-18979: Update Sidebar to match UI Kit

### PR DESCRIPTION
**Pull request by ashenden**
The DC/OS UI Kit has made a number of improvements and adjustments to the appearance of the sidebar, it's menus, and the fixed header. The styles that define the sidebar, as well as the variables that drive these styles, have been refactored to mirror the preferred appearance as well as make future maintenance of the sidebar easier.

**NOTE:** The header markup has been adjusted and requires involvement by @nLight to fix the behavior as requested and tracked in this JIRA ticket: https://jira.mesosphere.com/browse/DCOS_OSS-1758. Until the requests made in that ticket are resolved, the sidebar header dropdown menu will not appear for the application.

### CODE REVIEW

File: src/js/components/SidebarHeader.js [hunk 1]
```diff
@@ -141,7 +141,24 @@ class SidebarHeader extends mixin(StoreMixin) {
...
+              <div className="header-image" />
+            </div>
+            <div className="header-details">
+              <span className="header-title">
+                <span>
+                  {this.getClusterName()}
+                </span>
+              </span>
+              <span className="header-subtitle">
+                Firstname Lastname
```
Comment (MatApple) [inline src/js/components/SidebarHeader.js]: @ashenden is this a placeholder?

Comment (ashenden) [reply src/js/components/SidebarHeader.js]: @MatApple yes.  See my comments above about issue DCOS-18942. I'm currently using placeholder markup and content here that matches the preferred structure for the header and have commented out the old header.

Comment (ashenden) [reply src/js/components/SidebarHeader.js]: Also, I'm still working on resolving the tests that are failing.

**Comment by nLight** [comment]
Alfred test this

**Comment by ashenden** [comment]
retest this please

**Comment by ahoskins** [comment]
screen shot 2017-10-17 at 5 21 02 pm

**Comment by ahoskins** [comment]
Alfreddo test this

**Comment by ahoskins** [comment]
Looks like some spacing changed and we'll have to pass `.click({ force: true })` to where ever this is being clicked. Cypress is a baby and won't click if the element is partially covered.

**Comment by ahoskins** [comment]
Do we want this much margin on the left side of these items?
screen shot 2017-10-18 at 12 56 31 pm

**Comment by ashenden** [comment]
@MatApple yes.  At the moment. 

Is this ok to merge?

**Comment by nLight** [comment]
Hold merge as the SidebarDropdown is WIP

**Comment by ashenden** [comment]
@nLight following up on your comment above.  Are you actively working on the Sidebar Dropdown patch being tracked here https://jira.mesosphere.com/browse/DCOS_OSS-1758 ?

There a number of PRs that cascade as a dependency of this.  So, the sooner we can resolve that open issue, the sooner we can merge these.

**Comment by orlandohohmeier** [comment]
@ashenden no one is currently working on [DCOS_OSS-1758](https://jira.mesosphere.com/browse/DCOS_OSS-1758). We're going to pull that one into the next sprint if we have the capacity to do so. Please keep in mind that we have rather packed backlog and that the UI Kit alignment effort wasn't included in our initial capacity planning.

File: src/js/components/SidebarHeader.js (hunk 1 above)
Comment (nLight) [reply src/js/components/SidebarHeader.js]: Hey @ashenden! Sorry for takin that long, why cant we adjust `UserAccountDropdownTrigger` component? Isn't there everything in place already?

**Comment by nLight** [comment]
Hey @ashenden @MatApple, @orlandohohmeier!

I went ahead and added a feature to `Dropdown` https://github.com/mesosphere/reactjs-components/pull/418 to enable this change. We can do it even without messing with `reactjs-components` that would require us overriding all the `button` specific styles so it becomes a container. If you decide to go with css only solution feel free to reject https://github.com/mesosphere/reactjs-components/pull/418

**Comment by ashenden** [comment]
@MatApple @orlandohohmeier any thoughts on @nLight's comment above?

**Comment by MatApple** [comment]
@nLight @ashenden let's stick with solving via https://github.com/mesosphere/reactjs-components/pull/418 as overriding the button styles is not pretty and we need to adjust this to use custom trigger elements anyway.

**Comment by ashenden** [comment]
@MatApple thanks.  Anything I can do to help, or will you and @nLight sort this out?

@nLight I'm running your branch of `reactjs-components` (`drozhkov/dcos-18979-allow-custom-drop-down-trigger`) locally and followed the link instructions found here -- https://github.com/dcos/dcos-ui/blob/master/CONTRIBUTING.md#reactjs-components. However, I still see a `button` wrapping the markup (https://cl.ly/2D2a3S08272Z).  What am I doing wrong?

**Comment by MatApple** [comment]
@ashenden, there are some requested changes on the reactjs PR so we'll get those resolved, merge it, bump the release and then adjust this PR with the updated package version and @nLight will have to alter the code to use a custom `trigger` component in place of the default `ButtonTrigger`.

Once that's been updated you shouldn't see the button.

**Comment by ashenden** [comment]
@MatApple right, but I use `npm link` to point to a local checkout of `reactjs-components` with @nLight's branch checked-out.  Shouldn't this theoretically show me locally what things will look like once you've merged in the changes and bumped the release? (see comment above)

**Comment by nLight** [comment]
@ashenden that's correct you should see components from the work-in-progress branch. The only issue here is that the solution on the `reactjs-components` branch have diverged to support review comments, so the code I wrote here is broken for now. 

@juliangieseke picked up the task and we're collaborating with him to get it merged. So there's no question whether we will or won't update `reactjs-components`, we will. Right after that I'll update my solution in this branch.

**Comment by ashenden** [comment]
Previous check before last commit passed (https://jenkins.mesosphere.com/service/jenkins/job/Frontend/job/dcos-ui-pull-requests/1438/).  This new commit was adding an image.

**Comment by MatApple** [comment]
alfred retest this please

**Pull request merged**